lms/djangoapps/badges/models.py

```diff
@@ -0,0 +1,280 @@
+            u"that do not have a specified badge image. You can have only one default image."
+        )
+    )
+
+    def __unicode__(self):
+        return u"<CourseCompleteImageConfiguration for '{mode}'{default}>".format(
+            mode=self.mode,
+            default=u" (default)" if self.default else u''
+        )
+
+    def clean(self):
+        """
+        Make sure there's not more than one default.
+        """
+        # pylint: disable=no-member
+        if self.default and CourseCompleteImageConfiguration.objects.filter(default=True).exclude(id=self.id):
+            raise ValidationError(_(u"There can be only one default image."))
+
+    @classmethod
+    def image_for_mode(cls, mode):
+        """
+        Get the image for a particular mode.
+        """
+        try:
+            return cls.objects.get(mode=mode).icon
+        except cls.DoesNotExist:
+            # Fall back to default, if there is one.
+            return cls.objects.get(default=True).icon
+
+
+class CourseEventBadgesConfiguration(ConfigurationModel):
+    """
+    Determines the settings for meta course awards-- such as completing a certain
+    number of courses or enrolling in a certain number of them.
+    """
+    courses_completed = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the number of completed courses to award a badge for, a comma, and the slug of a "
+            u"badge class you have created with the issuing component 'edx__course'. "
+            u"For example: 3,course-v1:edx/Demo/DemoX"
+        )
+    )
+    courses_enrolled = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the number of enrolled courses to award a badge for, a comma, and the slug of a "
+            u"badge class you have created with the issuing component 'edx__course'. "
+            u"For example: 3,course-v1:edx/Demo/DemoX"
+        )
+    )
+    course_groups = models.TextField(
+        blank=True, default='',
+        help_text=_(
+            u"On each line, put the slug of a badge class you have created with the issuing component 'edx__course' to "
+            u"award, a comma, and a :-separated list of course keys that the user will need to complete to get this "
+            u"badge. For example: slug_for_compsci_courses_group_badge,course-v1:CompSci+Course+First,course-v1:"
+            u"CompsSci+Course+Second"
+        )
+    )
+
+    def __unicode__(self):
+        return u"<CourseEventBadgesConfiguration ({})>".format(u"Enabled" if self.enabled else u"Disabled")
+
+    @staticmethod
+    def get_specs(text):
+        """
+        Takes a string in the format of:
+            int,course_key
+            int,course_key
+
+        And returns a dictionary with the keys as the numbers and the values as the course keys.
+        """
+        specs = re.split('[\n\r]', text)
```

Comment:
Using `text.splitlines()` would be both more correct and simpler here.  (More correct because it handles DOS line endings correctly instead of resulting in empty ghost lines for each line ending.)